build-data-capturing-gradle-samples/capture-git-diffs/gradle-git-diffs.gradle

```diff
@@ -13,88 +13,103 @@ import java.util.concurrent.TimeUnit
  * In order for the gist to be created successfully, you must specify the Gradle property `gistToken` where
  * the value is a Github access token that has gist permission on the given Github repo.
  *
- * See https://docs.github.com/en/free-pro-team@latest/rest/reference/gists#create-a-gist for reference.
- */
+ * See https://docs.github.com/en/free-pro-team@latest/rest/reference/gists#create-a-gist for reference.*/
 
 def buildScanApi = project.extensions.findByName('buildScan')
 if (!buildScanApi) {
-    return
+  return
 }
 
+def capture = new Capture(gradle.startParameter.offline, gradle.startParameter.continuous, gradle.rootProject.logger, getProviders().gradleProperty('gistToken'), project.name)
+
 buildScan.background {
-    captureGitDiffInGist()
+  capture.captureGitDiffInGist(buildScanApi)
 }
 
-void captureGitDiffInGist() {
-    def isCapturingScan = !gradle.startParameter.offline && !gradle.startParameter.continuous
+class Capture {
+
+  def offline
+  def continuous
+  def logger
+  def gistTokenProvider
+  def projectName
+
+  Capture(offline, continuous, logger, gistTokenProvider, projectName) {
+    this.offline = offline
+    this.continuous = continuous
+    this.logger = logger
+    this.gistTokenProvider = gistTokenProvider
+    this.projectName = projectName
+  }
+
+  void captureGitDiffInGist(api) {
+    def isCapturingScan = !offline && !continuous
```

Comment:
can we move this checking logic outside the of the `Capture` class in order to simplify the logic?

Reply:
I extracted the `isEnabled` logic into its own method. This has to be inside the `Capture` class though, if not, we receive an exception while restoring the CC entry
`groovy.lang.MissingPropertyException: Could not get unknown property 'gradle' for BuildScanExtension`